plotly/figure_factory/_dendrogram.py

```diff
@@ -288,12 +290,11 @@ def get_dendrogram_traces(self, X, colorscale, distfun, linkagefun, hovertext):
             hovertext_label = None
             if hovertext:
                 hovertext_label = hovertext[i]
-            trace = dict(
-                type='scatter',
+            trace = graph_objs.Scatter(
                 x=np.multiply(self.sign[self.xaxis], xs),
                 y=np.multiply(self.sign[self.yaxis], ys),
                 mode='lines',
-                marker=dict(color=colors[color_key]),
+                marker=graph_objs.Marker(color=colors[color_key]),
```

Comment:
`graph_objs.Marker` is deprecated. So this should either stay a `dict`, or be `graph_objs.scatter.Marker`.  Given the dendrogram performance discussion in https://github.com/plotly/plotly.py/issues/1052, I think we should avoid using `graph_objs` here and stick with `dict`s.  This would also be consistent with my goals described in https://github.com/plotly/plotly.py/issues/1079